### ISSUE TITLE
BUG: in normal mode, keystrokes are leaking to the site

### ISSUE THREAD

**Issue by kalbasit** [brookhong/Surfingkeys#425]
I use Colemak keyboard layout so your `j` to scroll down is my `e`. For example, when scrolling down On Github, I often end up on the edit page (Github maps `e` to edit).

Steps to reproduce:
- In your Surfingkeys settings, map `e` to `j`: `map('e', 'j');`
- Go to https://github.com/brookhong/Surfingkeys/blob/master/README.md
- Hold `e` to scroll down

Expected:
Page keep scrolling down

Actual:
Page will scroll down but at some point you'll find yourself [here](https://github.com/brookhong/Surfingkeys/edit/master/README.md)

**Comment by brookhong** [comment]
I could not reproduce it. Anyone else could reproduce this?

**Comment by qwertea** [comment]
This is a pretty old issue so I'm not sure if I should start a new one instead, but I can reproduce it. Well, not on that page, but when I scroll down with j while watching a YouTube video, the video keeps seeking back 10 seconds. (j is the shortcut for that on YouTube.) It doesn't happen with every keystroke but eventually it will happen if you keep scrolling down. Also k to scroll up keeps pausing and resuming the video.

I can also reproduce this on reddit with the RES extension. RES has its own vim-like bindings and sometimes the keypresses leak through to it. First I thought that's just extension conflict but it's worth noting that neither of these things happen on Vimium or cVim.

**Comment by qwertea** [comment]
Can still reproduce on both Chromium or Firefox. I'm on Linux, using xorg. Posting because I figured out it only happens when the keyboard repeat rates are set high (in my case it's set to 60, if I set it down to 40 I'm no longer able to reproduce it)

**Comment by kalbasit** [comment]
@qwertea My repeat rate is set to 33 so not sure if that is the case, but I'm glad that I'm not alone in this

```sh-session
Keyboard Control:
  auto repeat:  on    key click percent:  0    LED mask:  00000000
  XKB indicators:
    00: Caps Lock:   off    01: Num Lock:    off    02: Scroll Lock: off
    03: Compose:     off    04: Kana:        off    05: Sleep:       off
    06: Suspend:     off    07: Mute:        off    08: Misc:        off
    09: Mail:        off    10: Charging:    off    11: Shift Lock:  off
    12: Group 2:     off    13: Mouse Keys:  off
  auto repeat delay:  200    repeat rate:  33
  auto repeating keys:  00ffffffdffffbbf
                        fadfffefffedffff
                        9fffffffffffffff
                        fff7ffffffffffff
  bell percent:  0    bell pitch:  400    bell duration:  100
Pointer Control:
  acceleration:  2/1    threshold:  4
Screen Saver:
  prefer blanking:  yes    allow exposures:  yes
  timeout:  600    cycle:  600
Colors:
  default colormap:  0x22    BlackPixel:  0x0    WhitePixel:  0xffffff
Font Path:
  /usr/share/fonts/TTF/,/usr/share/fonts/OTF/,/usr/share/fonts/100dpi/,/usr/share/fonts/75dpi/,built-ins
DPMS (Energy Star):
  Standby: 600    Suspend: 600    Off: 600
  DPMS is Enabled
  Monitor is On
```

**Comment by Monirzadeh** [comment]
I have the same problem
reproduce: open a youtube video while that play scrolling down with 'j' until the end of the comments that backward youtube video
it's so annoying :disappointed:

**Comment by Shamaoke** [comment]
Twitter has its own set of navigation hotkeys. For example, by pressing the `j` and `k` keys you can go to the next and previous tweets. When you press these keys, the tweet itself is highlighted by the blue frame and the page is scrolled to show the tweet, if it's hidden.

When using Surfingkeys on Twitter, the site and the addon use keystrokes cooperatively: sometimes a key is leaked to the site, sometimes it's not. Also, in some cases, Twitter grabs all of keystrokes and the addon completely stops to react on them.

**Comment by brookhong** [comment]
So everyone here with this issue are on Linux?

**Comment by Shamaoke** [comment]
@brookhong I am.

**Comment by kalbasit** [comment]
Linux here as well. Using Chromium 74.0.3729.157. I still have the issue.

**Comment by Monirzadeh** [comment]
> So everyone here with this issue are on Linux?

I don't read code yet but do you have any idea about which part maybe make this problem
I don't have any problem with vimium may be a side by side  comparison can be helpful

**Comment by brookhong** [comment]
@Monirzadeh You can start from https://github.com/brookhong/Surfingkeys/blob/master/content_scripts/normal.js#L110

**Comment by skaniol** [comment]
It also happens on  
```
Windows 10 Pro 1903 OS build 18362.449  
Brave Version 0.70.121 Chromium: 78.0.3904.70 (Official Build) (64-bit)
Surfingkeys 0.9.50
```
My keyboard repeat rate is maxed from Control Panel:
```bash
$ cat /proc/registry/HKEY_CURRENT_USER/Control\ Panel/Keyboard/KeyboardSpeed
31
```
```
C:\Users\skaniol>reg query "HKCU\Control Panel\Keyboard"

HKEY_CURRENT_USER\Control Panel\Keyboard
    KeyboardDelay    REG_SZ    1
    KeyboardSpeed    REG_SZ    31
    InitialKeyboardIndicators    REG_SZ    2
```

The problem doesn't occur if I use abnormally slow repeat rate, like 5 or lower. That's unusable.  

BTW, my CPU isn't very fast (Core 2 Duo T9300 @ 2.5 GHz), but this issue doesn't occur on other vim-like extensions. I guess Surfingkeys is heavier. Hopefully, it can be optimized. I don't have any custom mappings and customizations at the moment, beside disabling the smoothscroll.

**Comment by crocket** [comment]
I also experience the same issue on vivaldi which is a variant of chrome.

**Comment by crocket** [comment]
I confirmed that surfingkeys leaks keystrokes while javascript adds more HTML elements on the fly.

When I scroll down youtube comments, youtube automatically adds more comments through javascript.
When I scroll down list of items on a shopping website, it adds more items automatically when I scroll to the bottom.

A workaround is to disable surfingkeys and enable vimium on websites that add more HTML elements on the fly.

**Comment by crocket** [comment]
I discovered that keystrokes such as `j` and `k` reliably leak on http://prod.danawa.com/list/?cate=112782&15main_11_02

**Comment by aliveupstairs** [comment]
@brookhong, Here's a video:
[![image](https://user-images.githubusercontent.com/46059092/76655981-df934500-656e-11ea-997a-269f7e993c01.png)](https://i.imgur.com/0CK11yd.mp4)

**Comment by smhmd** [comment]
`j` and `k` are two vital keys in vim keybindings and scrollings is almost all you do in a browser. Could you please give this issue a higher priority, @brookhong? It's been open since 2017.

I really like the comprehensive array of possible commands and the interface. I **don't** want to use another extension, but using the mouse to scroll safely hinders my vim keys adoption in the browser. 

Thank you.

I, too, use Linux and `xset r rate 300 50`.

**Comment by brookhong** [comment]
Thanks @smhmd for the video, I have reproduced the issue on a virtual machine running ubuntu, will be fixed soon.

**Comment by brookhong** [comment]
Fixed in 0.9.60

The release for Firefox is published now, please have a try.
The release for Chrome is pending review on Chrome Web Store.
